rest_framework/permissions.py

```diff
@@ -172,7 +171,14 @@ def get_required_object_permissions(self, method, model_cls):
         return [perm % kwargs for perm in self.perms_map[method]]
 
     def has_object_permission(self, request, view, obj):
-        model_cls = view.queryset.model
+        try:
+            queryset = view.get_queryset()
+        except AssertionError as exc:
+            # view.get_queryset() didn't find .queryset
+            raise six.raise_from(AssertionError(
+                'Cannot apply DjangoObjectPermissions on a view that '
+                'does not have `.queryset` property nor redefines `.get_queryset()`.'), exc)
```

Comment:
Let's use:

> ... does not have a `.get_queryset()` method, or a '.queryset' property.

The "nor" is okay, but may be a bit obscure for some international readers.